### ISSUE TITLE
Instruction to compute ilog2 of a number

### ISSUE THREAD

**Issue by hackaugusto** [0xPolygonMiden/miden-vm#807]
It is possible to implement `ilog2` in pure MASM, but that has to be done with a loop over the bits, something like:

```
# stack: [number, ...] -> [ilog2, ...]
export.ilog2
  push.0          # bit position from the most-signficant, `31-bit_pos` equals to ilog2
  push.2147483648 # bitmask from high to low bit

  # find the first most-significant true bit
  dup.0 dup.3 u32checked_and eq.0 while.true
    div.2 swap add.1 swap
    dup.0 dup.3 u32checked_and eq.0
  end

  # compute ilog2
  drop sub.31 swap drop
end
```

It would be nice to have an instruction to compute the above.

**Comment by hackaugusto** [comment]
note: @bobbinth suggested using non-determinism input instead of an operation

**Comment by bobbinth** [comment]
Closed by #1176.
